### ISSUE TITLE
Difference between net5.0 and netframework 4.7.2 in test tear down

### ISSUE THREAD

**Issue by StarWars999123** [nunit/nunit#4688]
Hi, 

I noticed a difference during the tear down of netframework (4.7.2) and netcore (net5.0) test projects: 
My previously set variables are not longer accessible and the classes (e.g. a System.Configuration.ConfigurationElement with some temporary variables) get reinitialized. Theirby I loose right now some important information, that persists in netframework projects. 

I use specflow.Nunit on top, but I assume, that this is a behaviour difference cause by NUnit. 
The accessing TearDown method is the following Specflow hook: 
```
 [AfterScenario(Order = 1000)]
 public void AfterScenarioTestEnd(ScenarioContext scenarioContext, ISpecFlowOutputHelper _specFlowHelper)
 {}
```
and during debugging, it gets called with the following stack trace.
![image](https://github.com/nunit/nunit/assets/16309681/cb678833-b66b-4552-9a94-f62f5914e286)

I couldn't find any documentation so far about differences between netframework and netcore, but my tests, showed, that there is a difference in the behaviour. How should this be fixed? 

Thank you in advance :)

**Comment by manfred-brands** [comment]
@StarWars999123 I presume you are NUnit version 3.xx? Version 4.xx does not support .net core 5.0, only 6.0.
Do you see a different stack trace between .net core 5.0 and .net framework?

Do you use any (Default)TimeOut as that behaviour is different, although I don't see that in your stack trace.

**Comment by stevenaw** [comment]
@StarWars999123 Can you share a minimal repro? It should be as small as possible while still able to run "as-is" to make it easier for the team to investigate.
One option that makes it easy for us to investigate is to submit it as a PR to https://github.com/nunit/nunit/issues/4688

**Comment by StarWars999123** [comment]
Yes, I am using Nunit 3.13.3 still. Debugging showed a different time of initialization of those classes. I use it together with Specflow (3.9.74), but due to the initialization I would assume its still an Nunit topic. 
In the method given above I try to access the a path variable from a singleton instance (that might not be as single as the name suggests. see below.) 
From what I saw so far, I would assume that the singletons instance gets created ouf of two different contexts. The first instance for the test setup, where e.g. the Scenario name gets put into the requested variable and another instance in the afterscenario hook. 
With NET472 it looks like the instances for both contexts get initialized during startup of the test. That allows the variable containing the scenario name to be still set. 
While with NET5.0 it seems like the second instance gets created during shutdown, where the scenario name is not longer known. 

I can try to setup an example project, but this may take some time.

**Comment by stevenaw** [comment]
Thanks @StarWars999123 
Sorry, a bit of a poor link there in my previous message. The option I meant to suggest for a PR is to our "issues" repo: https://github.com/nunit/nunit.issues

A repro will help us understand a few things including how you're managing configuration, what runner you may be using (ex: console or the adapter + testhost, etc).

**Comment by StarWars999123** [comment]
> Thanks @StarWars999123 Sorry, a bit of a poor link there in my previous message. The option I meant to suggest for a PR is to our "issues" repo: https://github.com/nunit/nunit.issues
> 
> A repro will help us understand a few things including how you're managing configuration, what runner you may be using (ex: console or the adapter + testhost, etc).

Ok, did that. See: https://github.com/nunit/nunit.issues/pull/5

**Comment by OsirisTerje** [comment]
@StarWars999123 I tried your repo, and had to update it to the latest version of the packages.  (Might be something on my computer that is not correct for older packages).  Anyway, I did see the difference in the behaviour, but was not able to see anything related to NUnit there.  

Can you have another look?

What I got was:
<img width="230" alt="image" src="https://github.com/user-attachments/assets/5e79c506-66c5-4560-a805-76d6c06a1bd6">

**Comment by OsirisTerje** [comment]
I found this comment you added in the PR:

> Using the project you should be able to discover a difference during the initialization. While with Net4.7.2, a property gets initialized twice on startup, NET5.0/NET6.0 causes an initialization after the test already failed. As result, the variable stays empty.

Which property are you referring to?
 Can you please add an Assert for this?  Makes it easier to follow your reasoning.

**Comment by manfred-brands** [comment]
Your `BeforeScenario`:

```csharp
 var runningTest = SpecFlowTestConfig.Instance.RunningTest;
 runningTest.ScenarioLogPath = Path.Combine("Feature", scenarioTitle);
```

Your `AfterScenario`:

```csharp
 var instance = TestConfig.Instance;
 Console.WriteLine("RESULT ONLY WITH NET472 successful:" + Path.Combine(TestConfig.Instance.RunningTest.ScenarioLogPath, "SomeFile.log"));
```

You are setting `SpecFlowTestConfig.Instance.RunningTest.ScenarioLogPath` but then using `TestConfig.Instance.RunningTest.ScenarioLogPath` 

These are two different instances, so I would expect the latter to be `null`. Causing the exception in the failure shown.
They also use different sections: `"test"` vs `nameof(testSettings)`

So either this scenario is not matching the real one or we don't understand what you say goes wrong here.
